### PR TITLE
[ORCO-5] Bugfix/improve replication dump performance

### PR DESCRIPTION
**Pull request by cpjulia**
### Scope & Purpose

Improved performance in replication dump protocol by inserting arrays of documents instead of one document at a time and also not retrieving the document revision field when not needed.

- [ ] :hankey: Bugfix (requires CHANGELOG entry)
- [ ] :pizza: New feature (requires CHANGELOG entry, feature documentation and release notes)
- [x] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification
- [x] :book: CHANGELOG entry made

#### Backports:
Obs.: Probably will be backported to 3.9
- [ ] Backport for 3.9: *(Please link PR)*

#### Related Information

- [ ] Docs PR: 
- [ ] Enterprise PR:
- [x] GitHub issue / Jira ticket number: https://arangodb.atlassian.net/browse/ORCO-5?atlOrigin=eyJpIjoiNjc3OWU1ZTY1YWI2NDAxOGJmOWM4NDE4ZjkzMTkzNGMiLCJwIjoiaiJ9
- [ ] Design document: 

### Testing & Verification

- [ ] This change is a trivial rework / code cleanup without any test coverage.
- [x] The behavior in this PR was *manually tested*
- [ ] This change is already covered by existing tests, such as *(please describe tests)*.
- [ ] This PR adds tests that were used to verify all changes:
  - [ ] Added new C++ **Unit tests**
  - [ ] Added new **integration tests** (e.g. in shell_server / shell_server_aql)
  - [ ] Added new **resilience tests** (only if the feature is impacted by failovers)
- [ ] There are tests in an external testing repository:
- [ ] I ensured this code runs with ASan / TSan or other static verification tools

### CODE REVIEW

File: arangod/Replication/DatabaseInitialSyncer.cpp [hunk 1]
```diff
@@ -831,6 +874,21 @@ Result DatabaseInitialSyncer::fetchCollectionDump(arangodb::LogicalCollection* c
...
+      _config.applier._restrictType == ReplicationApplierConfiguration::RestrictType::Include &&
+      _config.applier._restrictCollections.size() == 1 &&
+      !hasDocuments(*coll)) {
+    // DB server doing shard synchronization. now try to fetch everything in a single VPack array.
+    // note: only servers >= 3.10 will honor this URL parameter. servers that are not capable of
+    // this format will simply ignore it and send the old format.
+    // the syncer has code to tell the two formats apart.
+    // note: we can only add this flag if we are sure there are no documents present locally.
+    // everything else is not safe.
+    baseUrl += "&single=true";
```
Comment (mpoeter) [inline arangod/Replication/DatabaseInitialSyncer.cpp:890]: I think the name `single` is rather confusing in this context. We either get a "single array" or a bunch of "single vpack slices". Without the comment it is not really clear which one `single=true` is used for. Can we rename it to `array`?

File: arangod/RocksDBEngine/RocksDBRestReplicationHandler.cpp [hunk 2]
```diff
@@ -755,8 +755,20 @@ void RocksDBRestReplicationHandler::handleCommandDump() {
...
+  // wrapped into a {"type":2300,"data":<document>} envelope.
   bool const useEnvelope = _request->parsedValue("useEnvelope", true);
 
+  // "single" URL parameter supported from >= 3.10 onwards. it defaults to "false"
+  // if not set. when explictly set to "true", we can get away with sending all
+  // documents as one big velocypack array, instead of sending multiple velocypack
+  // documents one following another. this has the advantage that on the client side
+  // we will receive a velocypack array which is ready to be fed into a multi-document
+  // operation.
+  bool const singleArray = _request->parsedValue("single", false);
```
Comment (mpoeter) [inline arangod/RocksDBEngine/RocksDBRestReplicationHandler.cpp:770]: See above.

**Review by mpoeter — APPROVED** [review]
I would prefer if we could rename the new URL parameter, otherwise LGTM,

File: arangod/Replication/DatabaseInitialSyncer.cpp (hunk 1 above)
Comment (jsteemann) [reply arangod/Replication/DatabaseInitialSyncer.cpp]: Renamed!

File: arangod/RocksDBEngine/RocksDBRestReplicationHandler.cpp (hunk 2 above)
Comment (jsteemann) [reply arangod/RocksDBEngine/RocksDBRestReplicationHandler.cpp]: Renamed!

**Pull request merged**